pygmt/datasets/load_remote_dataset.py

```diff
@@ -210,20 +210,22 @@ def _load_remote_dataset(
     The returned :class:`xarray.DataArray` doesn't support slice operation for
     tiled grids.
     """
-
-    if registration in ("pixel", "gridline", None):
+    dataset = datasets[dataset_name]
+    if resolution not in dataset.resolutions.keys():
+        raise GMTInvalidInput(f"Invalid resolution '{resolution}'.")
+    if registration is None:
+        registration = dataset.resolutions[resolution].registrations[0]
```

Comment:
Okay. I will be back on December 26 and came update the PR then.

Reply:
> In either case, I also think we should put `gridline` before `pixel`, because letter `g` is in front of letter `p` and `gridline` and `pixel` are internally represented as 0 and 1 in GMT.

I feel these changes should be done in a separate PR so that this PR is smaller and easier to review.